sparta/sparta/resources/Array.hpp

```diff
@@ -884,6 +899,33 @@ namespace sparta
             return aged_list_;
         }
 
+        /**
+         * \brief Update the relative age information for each entry. This is useful
+         * for the users to know the age of an entry. Age information has to be
+         * updated for all entries at once. But only needs to be updated when any
+         * entry is deallocated.
+         */
+        void updateRelativeAge_()
+        {
+            sparta_assert(ArrayT == ArrayType::AGED);
+
+            // Since our data_list_ always adds new items to the
+            // front.  The oldest data is actually kept at the end of
+            // the list.  We can iterate from the end to find the nth
+            // oldest item.
+            auto it = aged_list_.rbegin();
+            uint32_t idx = invalid_entry_;
+            for(uint32_t i = 0; i < aged_list_.size(); ++i)
```

Comment:
Since `aged_list_.size()` will not change in this loop, for debug builds, consider caching this value.

Reply:
Ok